### ISSUE TITLE
API to optimize table metadata

### ISSUE THREAD

**Issue by ajantha-bhat** [apache/iceberg#11425]
### Feature Request / Improvement

Table metadata has history of partition spec, sort orders, schemas. 

There is no way to shorten the history from the table metadata. It ends up bloating up the table metadata even though all the files are based on new spec, new sort order, new schema.

We need API that traverse all the reachable manifests and removes unused history from table metadata. 

### Query engine

None

### Willingness to contribute

- [x] I can contribute this improvement/feature independently
- [ ] I would be willing to contribute this improvement/feature with guidance from the Iceberg community
- [ ] I cannot contribute this improvement/feature at this time

**Comment by ajantha-bhat** [comment]
This one seems to be related: https://github.com/apache/iceberg/pull/10755, but it just handles the spec clean up. We need to handle the same for schema and sort orders.

**Comment by amogh-jahagirdar** [comment]
Thanks @ajantha-bhat , yeah so for this I've been working with @advancedxy and got busy and wasn't able to follow up here. One thing I've discussed with a few others at this is if we should just perform this metadata cleanup as part of snapshot expiration. See https://github.com/advancedxy/iceberg/pull/1

We should be able to extend this for schemas as well. Let me know what you think!

**Comment by ajantha-bhat** [comment]
yes, my idea was also to have it part of `expire_snapshots`. That would be nice.

**Comment by gaborkaszab** [comment]
Just FYI, that I started working on expiring schemas similarly as specs
